File: algorithms/data-structures/stack-operations/typescript/stackOperations.ts
export function stackOps(arr: number[]): number {
    if (arr.length === 0) return 0;
    const stack: number[] = [];
    const opCount = arr[0];
    let idx = 1, total = 0;
    for (let i = 0; i < opCount; i++) {
        const type = arr[idx], val = arr[idx + 1];
        idx += 2;
        if (type === 1) stack.push(val);
        else if (type === 2) total += stack.length > 0 ? stack.pop()! : -1;
    }
    return total;
}

File: algorithms/data-structures/stack-operations/python/stack_operations.py
def stack_ops(arr: list[int]) -> int:
    if not arr:
        return 0
    stack: list[int] = []
    op_count = arr[0]
    idx = 1
    total = 0
    for _ in range(op_count):
        op_type = arr[idx]
        val = arr[idx + 1]
        idx += 2
        if op_type == 1:
            stack.append(val)
        elif op_type == 2:
            if stack:
                total += stack.pop()
            else:
                total += -1
    return total
